src/backend/tests/partaj/core/test_api_referral.py

```diff
@@ -1387,5 +1413,182 @@ def test_unassign_unit_referral_from_invalid_state(self, _):
+
+        response = self.client.post(
+            f"/api/referrals/{referral.id}/change_urgencylevel/",
+            {
+                "urgencylevel_explanation": "La justification du changement.",
+                "urgencylevel": str(new_urgencylevel.id),
+            },
         )
+        self.assertEqual(response.status_code, 401)
+
+    def test_change_urgencylevel_by_random_logged_in_user(self, _):
+        """
+        Random logged-in users cannot change urgencylevel.
+        """
+        user = factories.UserFactory()
+        referral = factories.ReferralFactory()
+        new_urgencylevel = factories.ReferralUrgencyFactory()
+        self.assertNotEqual(new_urgencylevel.id, referral.urgency_level.id)
+
+        response = self.client.post(
+            f"/api/referrals/{referral.id}/change_urgencylevel/",
+            {
+                "urgencylevel_explanation": "La justification du changement.",
+                "urgencylevel": str(new_urgencylevel.id),
+            },
+            HTTP_AUTHORIZATION=f"Token {Token.objects.get_or_create(user=user)[0]}",
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_change_urgencylevel_unit_member(self, _):
+        """
+        A regular unit member cannot change urgencylevel.
+        """
+        referral = factories.ReferralFactory()
+        user = factories.UnitMembershipFactory(
+            role=models.UnitMembershipRole.MEMBER, unit=referral.units.get()
+        ).user
+        new_urgencylevel = factories.ReferralUrgencyFactory()
+        self.assertNotEqual(new_urgencylevel.id, referral.urgency_level.id)
+
+        response = self.client.post(
+            f"/api/referrals/{referral.id}/change_urgencylevel/",
+            {
+                "urgencylevel_explanation": "La justification du changement.",
+                "urgencylevel": str(new_urgencylevel.id),
+            },
+            HTTP_AUTHORIZATION=f"Token {Token.objects.get_or_create(user=user)[0]}",
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_change_urgencylevel_unit_admin(self, _):
+        """
+        A admin unit  member cann change urgencylevel.
+        """
+        referral = factories.ReferralFactory()
+        user = factories.UnitMembershipFactory(
+            role=models.UnitMembershipRole.ADMIN, unit=referral.units.get()
+        ).user
+        new_urgencylevel = factories.ReferralUrgencyFactory()
+        self.assertNotEqual(new_urgencylevel.id, referral.urgency_level.id)
+
+        response = self.client.post(
+            f"/api/referrals/{referral.id}/change_urgencylevel/",
+            {
+                "urgencylevel_explanation": "La justification du changement.",
+                "urgencylevel": str(new_urgencylevel.id),
+            },
+            HTTP_AUTHORIZATION=f"Token {Token.objects.get_or_create(user=user)[0]}",
+        )
+        self.assertEqual(response.status_code, 200)
```

Comment:
You could add a check on the referral that the urgency level is indeed changed to the new value